pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -756,3 +788,165 @@ func LookupNetworkByName(networks []k6tv1.Network, name string) *k6tv1.Network {
 	}
 	return nil
 }
+
+func loadVMLabelsConfiguration() {
+	virtClient, err := kubecli.GetKubevirtClient()
+	if err != nil {
+		log.Log.Errorf("vm-labels: failed to get KubeVirt client: %v", err)
+		resetVMLabelsConfigToDefaults()
+		return
+	}
+
+	namespace, err := util.GetNamespace()
+	if err != nil {
+		log.Log.Errorf("vm-labels: failed to determine namespace: %v", err)
+		resetVMLabelsConfigToDefaults()
+		return
+	}
+
+	configMap, err := virtClient.CoreV1().ConfigMaps(namespace).Get(
+		context.TODO(), vmLabelsConfigMapName, metav1.GetOptions{})
+	if err != nil {
+		resetVMLabelsConfigToDefaults()
+		return
+	}
+
+	updateVMLabelsConfigFromConfigMap(configMap)
+}
+
+func updateVMLabelsConfigFromConfigMap(configMap *k8sv1.ConfigMap) {
+	if configMap.Data == nil {
+		resetVMLabelsConfigToDefaults()
+		return
+	}
+
+	// Update under lock to avoid races
+	vmLabelsConfigMu.Lock()
+	defer vmLabelsConfigMu.Unlock()
+
+	if allowlistData, exists := configMap.Data["allowlist"]; exists {
+		allowlist := parseLabelsFromString(allowlistData)
+		if len(allowlist) > 0 {
+			vmLabelsAllowlist = allowlist
+		}
+	} else {
+		vmLabelsAllowlist = []string{"*"}
+	}
+
+	if ignorelistData, exists := configMap.Data["ignorelist"]; exists {
+		ignorelist := parseLabelsFromString(ignorelistData)
+		vmLabelsIgnorelist = ignorelist
+	} else {
+		vmLabelsIgnorelist = []string{}
+	}
+
+}
+
+func parseLabelsFromString(data string) []string {
+	if strings.TrimSpace(data) == "" {
+		return []string{}
+	}
+
+	labels := strings.Split(data, ",")
+	result := make([]string, 0, len(labels))
+
+	for _, label := range labels {
+		trimmed := strings.TrimSpace(label)
+		if trimmed != "" {
+			result = append(result, trimmed)
+		}
```

Comment:
Should we check for errors in case the labels were not added properly to the allow/deny lists?

Reply:
we agreed on empty allow list or non existing labels - metric will not be created and log.
ignore list will expect only exact label keys.

fixed.